pkg/operator/workloads/logs.go

```diff
@@ -235,57 +235,57 @@ func podCheck(podCheckChan chan struct{}, socketWriterError chan error, initialP
 				return
 			}
 
-			runningPodsMap := make(map[string]kcore.Pod)
+			latestRunningPodsMap := make(map[string]kcore.Pod)
 			latestRunningPods := strset.New()
 			for _, pod := range pods {
 				if k8s.GetPodStatus(&pod) != k8s.PodStatusPending {
 					latestRunningPods.Add(pod.GetName())
-					runningPodsMap[pod.GetName()] = pod
+					latestRunningPodsMap[pod.GetName()] = pod
 				}
 			}
 
-			runningPods := strset.New()
+			prevRunningPods := strset.New()
 
 			for podName := range processMap {
-				runningPods.Add(podName)
+				prevRunningPods.Add(podName)
 			}
 
-			newPods := strset.Difference(latestRunningPods, runningPods)
-			podsToDelete := strset.Difference(runningPods, latestRunningPods)
-			podsToKeep := strset.Intersection(runningPods, latestRunningPods)
+			newPods := strset.Difference(latestRunningPods, prevRunningPods)
+			podsToDelete := strset.Difference(prevRunningPods, latestRunningPods)
+			podsToKeep := strset.Intersection(prevRunningPods, latestRunningPods)
 
 			// Prioritize adding running pods
-			podsToAdd := []string{}
-			podsToAddP2 := []string{}
+			podsToAddRunning := []string{}
+			podsToAddNotRunning := []string{}
 			for podName := range newPods {
-				pod := runningPodsMap[podName]
+				pod := latestRunningPodsMap[podName]
 				if k8s.GetPodStatus(&pod) == k8s.PodStatusRunning {
-					podsToAdd = append(podsToAdd, podName)
+					podsToAddRunning = append(podsToAddRunning, podName)
 				} else {
-					podsToAddP2 = append(podsToAddP2, podName)
+					podsToAddNotRunning = append(podsToAddNotRunning, podName)
 				}
 			}
-			podsToAdd = append(podsToAdd, podsToAddP2...)
+			podsToAdd := append(podsToAddRunning, podsToAddNotRunning...)
 
-			for _, podName := range podsToAdd {
-				if len(podsToKeep) < maxParallelPodLogging {
-					process, err := startKubectlProcess(runningPodsMap[podName], previous, &os.ProcAttr{
+			numNewPods := maxParallelPodLogging - len(podsToKeep)
+
+			if numNewPods > 0 {
+				for _, podName := range podsToAdd[:numNewPods] {
```

Comment:
Can this be an index out of bounds on `podsToAdd`? e.g. if there was 1 pod and we added 1, `numNewPods` would be 4, and `podsToAdd` would have length 1, but we'd have `podsToAdd[:4]`?

Reply:
oh oops, thanks, mixed up behaviour with python